### ISSUE TITLE
Eloquent Adapter: HasMany sync/detach doesn't work with MorphMany relationships

### ISSUE THREAD

**Issue by flofloflo** [cloudcreativity/laravel-json-api#187]
The documentation says about Eloquent `morphMany` relationships that they are handled by the JSON API relationship type `hasMany`. This has the effect that the following exception is thrown when trying to update a resource with a  `morphMany` relationship:
```
Argument 1 passed to CloudCreativity\LaravelJsonApi\Eloquent\HasMany::sync() must be an instance of Illuminate\Database\Eloquent\Relations\HasMany, instance of Illuminate\Database\Eloquent\Relations\MorphMany given
```
The reason for this error is the type hinting to `Relations\HasMany` in the argument declaration of the [`sync`](https://github.com/cloudcreativity/laravel-json-api/blob/b91c1e559e70b8d6c3c44a789fd72410ca5c38c6/src/Eloquent/HasMany.php#L122) and [`detach`](https://github.com/cloudcreativity/laravel-json-api/blob/b91c1e559e70b8d6c3c44a789fd72410ca5c38c6/src/Eloquent/HasMany.php#L143) methods of the `CloudCreativity\LaravelJsonApi\Eloquent\HasMany` class.

@lindyhopchris Was this done by intention because the `sync`/`detach` only works with `hasMany` relationships or is it a bug?

**Comment by lindyhopchris** [comment]
@flofloflo no this should be working.

I had thought it was tested, but looking at the test that's only testing read situations, not write:
https://github.com/cloudcreativity/laravel-json-api/blob/master/tests/lib/Integration/Eloquent/MorphManyTest.php

I think probably what's happened here is I've forgotten to update the tests with the write scenarios when I was dev'ing the new relationship stuff.

**Comment by lindyhopchris** [comment]
Fixed and will be in the `1.0.0-beta.1` release.
